crates/ra_parser/src/grammar/patterns.rs

```diff
@@ -168,6 +168,7 @@ fn record_field_pat_list(p: &mut Parser) {
             T![.] if p.at(T![..]) => p.bump(T![..]),
 
             IDENT if p.nth(1) == T![:] => record_field_pat(p),
+            INT_NUMBER if p.nth(1) == T![:] => record_field_pat(p),
```

Comment:
I thought that I could use (IDENT  || INT_NUMBER) here but that didn't work.

Reply:
Yeah, rust support for patterns is annoyingly non-orthogonal....

Reply:
Shouldn't `IDENT | INT_NUMBER if p.nth(1) == T![:] => record_field_pat(p)` work?

Reply:
Huh... so it does. I could have sworn I tried the single `|` before. I'll send up a followup PR.

Reply:
> Hm, we already stabilized or patterns? Neat!

This was already supported, or patterns are like `Some(0 | 1)`, notice that the or is inside something else, not outside.